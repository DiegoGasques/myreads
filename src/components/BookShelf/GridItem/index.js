import React from "react";
import PropTypes from "prop-types";

function GridItem({ children }) {
  return <li className="grid-item">{children}</li>;
}

GridItem.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

export default GridItem;
