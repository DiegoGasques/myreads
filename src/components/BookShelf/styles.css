.book-shelf {
  background: floralwhite;
  padding: 20px 40px;
  border-radius: 30px;
  margin-bottom: 20px;
}

.book-shelf-title {
  margin: 0;
  padding-bottom: 10px;
  color: darkslategray;
}

.alice-carousel__next-btn {
  position: absolute;
  top: 62px;
  right: -37px;
  border-radius: 50%;
  width: 40px;
  height: 40px;
}

.alice-carousel__prev-btn-item,
.alice-carousel__next-btn-item {
  margin: 0;
  opacity: 0;
  z-index: 999999999;
  font-size: 1.5rem;
  position: absolute;
  top: 0;
  right: 0;
}

.alice-carousel__prev-btn {
  position: absolute;
  top: 62px;
  left: -37px;
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
