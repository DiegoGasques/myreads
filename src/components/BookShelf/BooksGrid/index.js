import React from "react";
import PropTypes from "prop-types";

import List from "../../List";
import "./styles.css";

function BooksGrid({ children }) {
  return <List className="books-grid-list">{children}</List>;
}

BooksGrid.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

export default BooksGrid;
