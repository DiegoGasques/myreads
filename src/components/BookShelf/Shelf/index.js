import React, { Component } from "react";
import PropTypes from "prop-types";
import Carousel from "react-alice-carousel";

import "react-alice-carousel/lib/alice-carousel.css";
import "./styles.css";

class Shelf extends Component {
  responsive = {
    0: { items: 1 },
    600: { items: 2 },
    950: { items: 3 },
    1324: { items: 4 }
  };

  handleOnDragStart = e => e.preventDefault();

  render() {
    const { children } = this.props;
    return (
      <div className="shelf">
        <i class="fas fa-angle-left" />
        <Carousel
          mouseDragEnabled
          items={children(this.handleOnDragStart)}
          responsive={this.responsive}
        />
        <i class="fas fa-angle-right" />
      </div>
    );
  }
}

Shelf.propTypes = {
  children: PropTypes.node
};

export default Shelf;
