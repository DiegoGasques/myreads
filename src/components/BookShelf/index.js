import React from "react";
import PropTypes from "prop-types";

import Shelf from "./Shelf";
import BooksGrid from "./BooksGrid";
import GridItem from "./GridItem";
import BookCard from "../BookCard";

import "./styles.css";

function BookShelf({ books }) {
  return (
    <div className="book-shelf">
      <h2 className="bookshelf-title">Currently Reading</h2>
      <Shelf>
        <BooksGrid>
          {books.map(b => (
            <GridItem>
              <BookCard book={b} />
            </GridItem>
          ))}
        </BooksGrid>
      </Shelf>
    </div>
  );
}

BookShelf.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default BookShelf;
