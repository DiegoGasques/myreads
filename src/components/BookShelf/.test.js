import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import BookShelf from "./index";
import { books } from "../../testFixtures";

it("Should match the component snapshot", () => {
  const wrapper = shallow(
    <BookShelf title="currentlyReading" filteredBooks={[books[0], books[1]]} />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
