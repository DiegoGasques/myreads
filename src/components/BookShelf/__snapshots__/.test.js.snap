// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="book-shelf"
>
  <h2
    className="book-shelf-title"
  >
    currentlyReading
  </h2>
  <Shelf>
    <Component />
  </Shelf>
</div>
`;
